internal/k8s/resource.go

```diff
@@ -0,0 +1,347 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/discovery/cached/disk"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/restmapper"
+	"k8s.io/client-go/util/homedir"
+)
+
+type Client struct {
+	dclient   *dynamic.DynamicClient
+	clientset *kubernetes.Clientset
+	rmapper   meta.RESTMapper
+	dc        *discovery.DiscoveryClient
+}
+
+type Resource struct {
+	Manifest           *unstructured.Unstructured
+	Children           []*Resource
+	LatestEventMessage string
+}
+
+// Returns resource kind as string
+func (r *Resource) GetKind() string {
+	return r.Manifest.GetKind()
+}
+
+// Returns resource name as string
+func (r *Resource) GetName() string {
+	return r.Manifest.GetName()
+}
+
+// Returns resource namespace as string
+func (r *Resource) GetNamespace() string {
+	return r.Manifest.GetNamespace()
+}
+
+// Returns resource apiversion as string
+func (r *Resource) GetApiVersion() string {
+	return r.Manifest.GetAPIVersion()
+}
+
+// This function takes a certain conditionType as input e.g. "Ready" or "Synced"
+// Returns the Status of the map with the conditionType as string
+func (r *Resource) GetConditionStatus(conditionKey string) string {
+	conditions, _, _ := unstructured.NestedSlice(r.Manifest.Object, "status", "conditions")
+	for _, condition := range conditions {
+		conditionMap, _ := condition.(map[string]interface{})
+		conditionType, _ := conditionMap["type"].(string)
```

Comment:
You're right I added a check to all assertions.
`type` should be fine ([source](https://github.com/kubernetes/apimachinery/blob/master/pkg/apis/meta/v1/types.go#L1507)), still added the check in case the `type` field might change in the future